packages/cli/src/modules/data-table/data-store.service.ts

```diff
@@ -373,4 +381,94 @@ export class DataStoreService {
 			}
 		}
 	}
+
+	private async validateDataTableSize() {
+		const now = new Date();
+
+		// If there's already a pending check, wait for it to complete
+		if (this.pendingSizeCheck) {
+			await this.pendingSizeCheck;
+			return;
+		}
+
+		// Check if we need to run the size check
+		const shouldRunCheck =
+			!this.lastCacheSizeCheck || now.getTime() - this.lastCacheSizeCheck.getTime() >= 3 * 1000;
+
+		if (shouldRunCheck) {
+			// Create and store the promise to prevent concurrent checks
+			this.pendingSizeCheck = this.performSizeCheck(now);
+
+			try {
+				await this.pendingSizeCheck;
+			} finally {
+				// Clear the pending check once it's done
+				this.pendingSizeCheck = null;
+			}
+		}
+	}
+
+	private async performSizeCheck(checkTime: Date): Promise<void> {
+		const maxSize = this.globalConfig.datatable.maxSize;
+
+		this.lastCacheSizeCheck = checkTime;
+
+		const currentSizeInMbs = await this.findDataTablesSize();
+
+		if (currentSizeInMbs >= maxSize) {
+			throw new DataStoreValidationError(
+				`Data store size limit exceeded: ${currentSizeInMbs}MB used, limit is ${this.globalConfig.datatable.maxSize}MB`,
+			);
+		}
```

Comment:
Here is where we will send the email. Wanted to keep a state in the class, but I guess this won't work for queue mode as we could then be in a case where we send the email letting the user know the warning threshold more than once. So I guess we need to keep the state in the database to avoid this. Maybe the settings table? 🤔 

```
	if (currentSizeInMbs >= warningThreshold) {

		}

```

Reply:
Yes, wanted to store in the settings table, as keeping the flag in memory would not work when working on queue mode.

Reply:
it wouldn't work on a crash either, seems like some place to persist on the db is necessary :/
maybe some logic in a transaction to read the current "last warning email sent" value, check it and in the same transaction write the current timestamp there, commit, and then to send the email could work?